contracts/NestedAsset.sol

```diff
@@ -42,7 +51,16 @@ contract NestedAsset is ERC721Enumerable, OwnableFactoryHandler {
     /// @return The token Uniform Resource Identifier (URI)
     function tokenURI(uint256 _tokenId) public view virtual override returns (string memory) {
         require(_exists(_tokenId), "URI query for nonexistent token");
-        return _tokenURIs[_tokenId];
+        if (isRevealed) {
+            return super.tokenURI(_tokenId);
+        } else {
+            return unrevealedTokenUri;
+        }
+    }
+
+    /// @inheritdoc ERC721
+    function _baseURI() internal view override returns (string memory) {
+        return baseUri;
```

Comment:
❔ Is this function really necessary? If used by the ERC721 contract, shouldn't it be declared in it already?